### PR TITLE
Make `generate_source` macro return case-sensitive names

### LINKED ISSUE #112: Names are automatically casted to lower-case, regardless of capitalisation in database
### Describe the bug
Source, schema, database, table, and column names are automatically casted to lowercase when generating source yaml files, regardless of the casing of the name in the actual database.

### Steps to reproduce
Generate a source yaml file from any database that contains capitalisation in the names of any of the aforementioned.

### Expected results
Conservation of capitalisation in the yaml file as it is in the database.

### Actual results
All-lowercase names.

### Are you interested in contributing the fix?
Yes, as it would just require minor adjustments of a few lines.

### PR DESCRIPTION
**Pull request by jgillies**
resolves #112 

This is a:
- [ ] documentation update
- [x] bug fix with no breaking changes
- [ ] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation
For organizations with strict standards around the letter case of their tables and columns. The existing behavior of the `generate_source` macro is that it returns all table and column names in lowercase. This makes it less useful for the previously mentioned organizations.

This adds the parameters `case_sensitive_tables` and `case_sensitive_cols` to the `generate_source` macro. Both default to `false` to preserve existing behavior.

## Checklist
- [ ] This code is associated with an issue which has been triaged and [accepted for development](https://docs.getdbt.com/docs/contributing/oss-expectations#pull-requests). 
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Review by qgab-flowdesk — APPROVED** [review]

File: macros/generate_source.sql [hunk 1]
```diff
@@ -44,7 +44,7 @@
 {% endif %}
 
 {% for table in tables %}
-    {% do sources_yaml.append('      - name: ' ~ table | lower ) %}
+    {% do sources_yaml.append('      - name: ' ~ (table if case_sensitive_tables else table | lower)) %}
```
Comment (Nipsuli) [inline macros/generate_source.sql:47]: Would it make sense to have the name still as lower case but add the uncased as `identifier`?

Comment (jgillies) [reply macros/generate_source.sql:47]: Ah, I never knew about the distinction between `name` and `identifier` is sources. Makes sense to use `identifier`!

Comment (jgillies) [reply macros/generate_source.sql:47]: Actually, I believe this is assigning the relation's `identifier` to the `table` variable, so no change would be necessary to pick up the logic you're suggesting. Does that make sense?

https://github.com/dbt-labs/dbt-codegen/blob/af12570eb8618b71f22c8ad77593d25a33dacd26/macros/generate_source.sql#L10

**Comment by gwenwindflower** [comment]
Hey @jgillies -- thanks for your patience with this, we were a bit short staffed while I was on medical leave for several months. This looks awesome and definitely should be added, thank you! 🙏🏻 

Couple things for consideration:
- Did you run the test suite? It looks like no tests were edited in this PR and a couple are failing now, would be worth running `make test` and checking those out, and perhaps looking at the other tests that start with `test_generate_source`. Don't hesitate to let me know if you get stuck with setting it up and running the tests, it can be a little confusing.
- Would you be up for expanding this to include Databases and Schemas? I think they should also probably be made optionally case sensitive.

Lastly, as you opened this quite a while ago, if your capacity has changed and you don't have the time to get into the above, no worries at all, just let me know and I can see about pushing some more commits to this branch. Would prefer to help you get this across the finish line yourself though if you're in for it!

Here to help with anything or any questions, just @ me.

**Comment by gwenwindflower** [comment]
Also, I went ahead and resolved the conflicts with main that have cropped up in the months this was waiting.

**Comment by jgillies** [comment]
> Hey @jgillies -- thanks for your patience with this, we were a bit short staffed while I was on medical leave for several months. This looks awesome and definitely should be added, thank you! 🙏🏻
> 
> Couple things for consideration:
> 
> * Did you run the test suite? It looks like no tests were edited in this PR and a couple are failing now, would be worth running `make test` and checking those out, and perhaps looking at the other tests that start with `test_generate_source`. Don't hesitate to let me know if you get stuck with setting it up and running the tests, it can be a little confusing.
> * Would you be up for expanding this to include Databases and Schemas? I think they should also probably be made optionally case sensitive.
> 
> Lastly, as you opened this quite a while ago, if your capacity has changed and you don't have the time to get into the above, no worries at all, just let me know and I can see about pushing some more commits to this branch. Would prefer to help you get this across the finish line yourself though if you're in for it!
> 
> Here to help with anything or any questions, just @ me.

Hi @gwenwindflower! I'm not able to jump back into this at the moment, so if someone else can push it across the finish line I'm totally OK with that. Otherwise, I can spend some time on it within the next couple months.

**Comment by gwenwindflower** [comment]
All good @jgillies -- thanks for letting us know, we've got it! Appreciate the work, patience, and thought to get it to here! ❤️

**Comment by pnadolny13** [comment]
@jgillies @gwenwindflower glad to see this PR! I've just run into the same issue.

Is there anything I can help with to get this over the finish line? I'm happy to contribute! 😄

**Comment by gwenwindflower** [comment]
> @jgillies @gwenwindflower glad to see this PR! I've just run into the same issue.
> 
> Is there anything I can help with to get this over the finish line? I'm happy to contribute! 😄

that would be super welcome! [take a look here](https://github.com/dbt-labs/dbt-codegen/tree/main/integration_tests) for how package development works and if that all makes sense, then the things remaining are:

- adding column sensitivity to other properties as listed above
- adding tests

very happy to answer any questions!

**Comment by pnadolny13** [comment]
@gwenwindflower I created a new PR https://github.com/dbt-labs/dbt-codegen/pull/168 that should have all the remaining items, let me know if you have any feedback!

Thanks @jgillies for creating this PR! If its ok with you then I think we can close this and get https://github.com/dbt-labs/dbt-codegen/pull/168 across the finishline 😄 .

**Comment by gwenwindflower** [comment]
Closed because of #168

**Pull request closed without merging**